### ISSUE TITLE
train DOTA-v2

### ISSUE THREAD

**Issue by Proient** [open-mmlab/mmrotate#196]
When training dotav2 dataset, the value of map is 0.1 ~ 0.2 and 12 epochs each time, using the official default network structure.The value of loss changes little

**Comment by zytx121** [comment]
Hi @Proient 
We haven't tested MMRotate on DOTA-v2.0. 

@yangxue0827  Do you have any suggestions?

**Comment by yangxue0827** [comment]
The amount of information provided is too limited to analyze.

**Comment by Proient** [comment]
> The amount of information provided is too limited to analyze.
Changes
1.	The config file  use is config/rotated_retinanet/rotated_retinanet_obb_r50_fpn_1x_dota_le90.py . I didn't create the custom.py , the changes I made directly on the source file ,such as config/_base_/datasets/dotav1.py
2.	In dotav1.py ,I put 'dota_ root','ann_file', and 'img_prefix'  changed to my own path. I created a new classes according to the dotav2 dataset,the num of class is 18.I append the class of  'airport','helipad' and 'container-crane'. I have not change the schedule_1x.py and the runtime.py.
3.	In 'mmrotate/datasets/dota.py ,my change is in the line 33 ,I also append the class of  'airport','helipad'and'container-crane'.
4.	In 'configs/rotated_retinanet/rotated_retinanet_obb_r50_fpn_1x_dota_le90.py ,my change is in the 'bbox_head','num_classes=18'.
Dataset
5.	I use the doatv2,I used all the images in part4,part5andpart6 as my training set,and I used the all txt files in 'lableTtx-v2.0/DOTA-v2.0_train' as the annfiles.In the 'val/images/' all images in part2 are used as val,and the annfiles also used lableTtx-v2.0/DOTA-v2.0_val'  (OBB)
6.	 I used 'python tools/data/dota/split/img_split.py --base-json tools/data/dota/split/split_configs/ss_train.json' to split dotav2-train images. Use 'python tools/data/dota/split/img_split.py --base-json tools/data/dota/split/split_configs/ss_val.json'to split dotav2-val images. After spilting the data, there are 8379 items in trian/annfiles folder and train/images folder and are 1439 items in val/annfiles folder and val/images folder.
7.	I used 'python tools/train.py  configs/rotated_retinanet/rotated_retinanet_obb_r50_fpn_1x_dota_le90.py' to train the dataset.
8.	The train epochs are 12.
9.	I try many times,the mAP always keep in the 0.1~0.2,andIt doesn't work well when used the 'latest.pth' to test  images .
10.	My environment is ubuntu20.04 ,RTX3090, mmrotate v0.2.0.
I hope I can get some help. Thank you very much!
 
Dotav1.py
# dataset settings
dataset_type = 'DOTADataset'
**classes=('plane', 'baseball-diamond', 'bridge', 'ground-track-field',
               'small-vehicle', 'large-vehicle', 'ship', 'tennis-court',
               'basketball-court', 'storage-tank', 'soccer-ball-field',
               'roundabout', 'harbor', 'swimming-pool', 'helicopter','airport','helipad','container-crane')**
# data_root = '../datasets/split_1024_dota1_0/'
data_root = 'data/split_ss_dota/'
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(type='RResize', img_scale=(1024, 1024)),
    dict(type='RRandomFlip', flip_ratio=0.5),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='Pad', size_divisor=32),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(1024, 1024),
        flip=False,
        transforms=[
            dict(type='RResize'),
            dict(type='Normalize', **img_norm_cfg),
            dict(type='Pad', size_divisor=32),
            dict(type='DefaultFormatBundle'),
            dict(type='Collect', keys=['img'])
        ])
]
data = dict(
    samples_per_gpu=2,
    workers_per_gpu=2,
    train=dict(
        type=dataset_type,
        **classes = classes,
        ann_file=data_root + 'train/annfiles/',
        img_prefix=data_root + 'train/images/',**
        pipeline=train_pipeline),
    val=dict(
        type=dataset_type,
        **classes = classes,
        ann_file=data_root + 'val/annfiles/',
        img_prefix=data_root + 'val/images/',**
        pipeline=test_pipeline),
    test=dict(
        type=dataset_type,
        **classes = classes,
        ann_file=data_root + 'test/iages/',
        img_prefix=data_root + 'test/images/',**
        pipeline=test_pipeline))
 
dota.py
# Copyright (c) OpenMMLab. All rights reserved.
 
 
@ROTATED_DATASETS.register_module()
class DOTADataset(CustomDataset):
    """DOTA dataset for detection.
 
    Args:
        ann_file (str): Annotation file path.
        pipeline (list[dict]): Processing pipeline.
        version (str, optional): Angle representations. Defaults to 'oc'.
        difficulty (bool, optional): The difficulty threshold of GT.
    """
    **CLASSES = ('plane', 'baseball-diamond', 'bridge', 'ground-track-field',
               'small-vehicle', 'large-vehicle', 'ship', 'tennis-court',
               'basketball-court', 'storage-tank', 'soccer-ball-field',
               'roundabout', 'harbor', 'swimming-pool', 'helicopter','airport','helipad','container-crane')**
 
   
rotated_retinanet_obb_r50_fpn_1x_dota_le90.py
_base_ = [
    '../_base_/datasets/dotav1.py', '../_base_/schedules/schedule_1x.py',
    '../_base_/default_runtime.py'
]
 
angle_version = 'le90'
model = dict(
    type='RotatedRetinaNet',
    backbone=dict(
        type='ResNet',
        depth=50,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=1,
        zero_init_residual=False,
        norm_cfg=dict(type='BN', requires_grad=True),
        norm_eval=True,
        style='pytorch',
        init_cfg=dict(type='Pretrained', checkpoint='torchvision://resnet50')),
    neck=dict(
        type='FPN',
        in_channels=[256, 512, 1024, 2048],
        out_channels=256,
        start_level=1,
        add_extra_convs='on_input',
        num_outs=5),
    bbox_head=dict(
        type='RotatedRetinaHead',
        # new version class is 18 ,append the 'airport','helipad','container-crane'
        **num_classes=18,**
        in_channels=256,
        stacked_convs=4,
        feat_channels=256,
        assign_by_circumhbbox=None,
        anchor_generator=dict(
            type='RotatedAnchorGenerator',
            octave_base_scale=4,
            scales_per_octave=3,
            ratios=[1.0, 0.5, 2.0],
            strides=[8, 16, 32, 64, 128]),
        bbox_coder=dict(
            type='DeltaXYWHAOBBoxCoder',
            angle_range=angle_version,
            norm_factor=None,
            edge_swap=True,
            proj_xy=True,
            target_means=(.0, .0, .0, .0, .0),
            target_stds=(1.0, 1.0, 1.0, 1.0, 1.0)),
        loss_cls=dict(
            type='FocalLoss',
            use_sigmoid=True,
            gamma=2.0,
            alpha=0.25,
            loss_weight=1.0),
        loss_bbox=dict(type='L1Loss', loss_weight=1.0)),
    train_cfg=dict(
        assigner=dict(
            type='MaxIoUAssigner',
            pos_iou_thr=0.5,
            neg_iou_thr=0.4,
            min_pos_iou=0,
            ignore_iof_thr=-1,
            iou_calculator=dict(type='RBboxOverlaps2D')),
        allowed_border=-1,
        pos_weight=-1,
        debug=False),
    test_cfg=dict(
        nms_pre=2000,
        min_bbox_size=0,
        score_thr=0.05,
        nms=dict(iou_thr=0.1),
        max_per_img=2000))
 
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(type='RResize', img_scale=(1024, 1024)),
    dict(
        type='RRandomFlip',
        flip_ratio=[0.25, 0.25, 0.25],
        direction=['horizontal', 'vertical', 'diagonal'],
        version=angle_version),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='Pad', size_divisor=32),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
]
data = dict(
    train=dict(pipeline=train_pipeline, version=angle_version),
    val=dict(version=angle_version),
    test=dict(version=angle_version))

**Comment by yangxue0827** [comment]
![image](https://user-images.githubusercontent.com/29257168/162467340-7b7ddef1-6a8a-4b74-8de5-32b1a4121cee.png)

**Comment by yangxue0827** [comment]
Your training and testing datasets may be incomplete.
